src/sidebar/services/groups.js

```diff
@@ -137,13 +138,20 @@ function groups($rootScope, store, api, isSidebar, localStorage, serviceUrl, ses
         params.document_uri = uri;
       }
       documentUri = uri;
-
-      // Fetch groups from the API.
-      return api.groups.list(params, null, { includeMetadata: true });
-    }).then(({ data, token }) => {
-      const isLoggedIn = token !== null;
+      
+      if (features.flagEnabled('community_groups')) {
+        let userGroups = api.profile.groups.read({expand: 'organization'});
+        let contextGroups = api.groups.list(params);
+        return Promise.all([userGroups, contextGroups]).then(combineGroups);
+      }
+      else {
+        // Fetch groups from the API.
+        return api.groups.list(params);
+      }
+    }).then(groups => {
+      const isLoggedIn = store.profile().userid !== null;
       const directLinkedAnnotation = settings.annotations;
-      return filterGroups(data, isLoggedIn, directLinkedAnnotation);
+      return filterGroups(groups, isLoggedIn, directLinkedAnnotation);
```

Comment:
We could move the filterGroups and injectOrganizations functions out of here and into the group util module as well but I didn't want to make this PR too big.